File: tests/algorithms/transpiler/FS/conversions/decimal_to_octal.fs
// Generated 2025-08-07 10:31 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let rec _str v =
    let s = sprintf "%A" v
    s.Replace("[|", "[")
     .Replace("|]", "]")
     .Replace("; ", " ")
     .Replace(";", "")
     .Replace("\"", "")
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec int_pow (``base``: int) (exp: int) =
    let mutable __ret : int = Unchecked.defaultof<int>
    let mutable ``base`` = ``base``
    let mutable exp = exp
    try
        let mutable result: int = 1
        let mutable i: int = 0
        while i < exp do
            result <- result * ``base``
            i <- i + 1
        __ret <- result
        raise Return
        __ret
    with
        | Return -> __ret
let rec decimal_to_octal (num: int) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable num = num
    try
        if num = 0 then
            __ret <- "0o0"
            raise Return
        let mutable octal: int = 0
        let mutable counter: int = 0
        let mutable value: int = num
        while value > 0 do
            let remainder: int = ((value % 8 + 8) % 8)
            octal <- octal + (remainder * (int_pow (10) (counter)))
            counter <- counter + 1
            value <- value / 8
        __ret <- "0o" + (_str (octal))
        raise Return
        __ret
    with
        | Return -> __ret
printfn "%s" (decimal_to_octal (2))
printfn "%s" (decimal_to_octal (8))
printfn "%s" (decimal_to_octal (65))
printfn "%s" (decimal_to_octal (216))
printfn "%s" (decimal_to_octal (512))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

File: tests/algorithms/x/Java/conversions/decimal_to_octal.java
public class Main {

    static int int_pow(int base, int exp) {
        int result = 1;
        int i = 0;
        while (i < exp) {
            result = result * base;
            i = i + 1;
        }
        return result;
    }

    static String decimal_to_octal(int num) {
        if (num == 0) {
            return "0o0";
        }
        int octal = 0;
        int counter = 0;
        int value = num;
        while (value > 0) {
            int remainder = Math.floorMod(value, 8);
            octal = octal + remainder * int_pow(10, counter);
            counter = counter + 1;
            value = value / 8;
        }
        return "0o" + _p(octal);
    }
    public static void main(String[] args) {
        System.out.println(decimal_to_octal(2));
        System.out.println(decimal_to_octal(8));
        System.out.println(decimal_to_octal(65));
        System.out.println(decimal_to_octal(216));
        System.out.println(decimal_to_octal(512));
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        return String.valueOf(v);
    }
}
